RadarSDK/RadarLocationManager.m

```diff
@@ -411,12 +451,82 @@ - (void)updateTrackingFromMeta:(RadarMeta *_Nullable)meta {
+- (void)changeRampedState:(BOOL)rampedUp {
+    [RadarSettings setRampedUp:rampedUp];
+    
+    StateChange *change = [[StateChange alloc] init];
+    change.timestamp = [NSDate date]; // Current timestamp
+    change.rampedUp = rampedUp;
+    [self.stateChanges addObject:change];
+}
+
+- (NSDictionary *)calculateRampedUpTimesAndCleanup {
+    NSDate *now = [NSDate date];
+    NSDate *oneHourAgo = [now dateByAddingTimeInterval:-3600]; // 1 hour ago
+    NSDate *twelveHoursAgo = [now dateByAddingTimeInterval:-43200]; // 12 hours ago
+    
+    NSTimeInterval totalRampedUpTimeOneHour = 0;
+    NSTimeInterval totalRampedUpTimeTwelveHours = 0;
+    NSDate *lastRampedUpStartOneHour = nil;
+    NSDate *lastRampedUpStartTwelveHours = nil;
+    NSMutableArray *validChanges = [NSMutableArray array]; // Array to store valid entries
+
+    for (StateChange *change in self.stateChanges) {
+        if ([change.timestamp compare:twelveHoursAgo] == NSOrderedAscending) {
+            continue; // Skip and effectively remove changes older than twelve hours
+        }
+
+        // Adding valid entries to the new array
+        [validChanges addObject:change];
+
+        // Calculation for the past 12 hours
+        if (change.rampedUp) {
+            lastRampedUpStartTwelveHours = change.timestamp;
+        } else if (lastRampedUpStartTwelveHours) {
+            totalRampedUpTimeTwelveHours += [change.timestamp timeIntervalSinceDate:lastRampedUpStartTwelveHours];
+            lastRampedUpStartTwelveHours = nil;
+        }
+
+        // Additional check for the past hour
+        if ([change.timestamp compare:oneHourAgo] != NSOrderedAscending) {
+            if (change.rampedUp) {
+                lastRampedUpStartOneHour = change.timestamp;
+            } else if (lastRampedUpStartOneHour) {
+                totalRampedUpTimeOneHour += [change.timestamp timeIntervalSinceDate:lastRampedUpStartOneHour];
+                lastRampedUpStartOneHour = nil;
+            }
+        }
+    }
+
+    // Final adjustments if the last state was ramped up and hasn't changed since
+    if (lastRampedUpStartTwelveHours) {
+        totalRampedUpTimeTwelveHours += [now timeIntervalSinceDate:lastRampedUpStartTwelveHours];
+    }
+    if (lastRampedUpStartOneHour) {
+        totalRampedUpTimeOneHour += [now timeIntervalSinceDate:lastRampedUpStartOneHour];
+    }
+
+    // Update the stateChanges array with the valid changes
+    self.stateChanges = validChanges;
+
+    return @{@"OneHour": @(totalRampedUpTimeOneHour), @"TwelveHours": @(totalRampedUpTimeTwelveHours)};
+}
+
+
 - (void)restartPreviousTrackingOptions {
     RadarTrackingOptions *previousTrackingOptions = [RadarSettings previousTrackingOptions];
     [[RadarLogger sharedInstance] logWithLevel:RadarLogLevelDebug message:@"Restarting previous tracking options"];
 
     if (previousTrackingOptions) {
         [Radar startTrackingWithOptions:previousTrackingOptions];
+        if ([RadarSettings rampedUp]) {
```

Comment:
this is block here is pretty complicated (in a part of the code that's already kinda tricky)- don't think we want to be special casing this this much

Reply:
Yes, good call. This can and should be removed.